libs/azure/odc/azure/__init__.py

```diff
@@ -0,0 +1,56 @@
+"""Azure blob storage crawling and YAML fetching utilities
+"""
+from azure.storage.blob import ContainerClient, BlobClient
+from typing import List, Tuple, Optional
+from multiprocessing.dummy import Pool as ThreadPool
+from functools import partial
+
+def find_blobs(account_url: str, container_name: str, credential: str, prefix: str, suffix: str):
+    blob_list = []
+    container = ContainerClient(account_url=account_url, container_name=container_name, credential=credential)
+    for blob_record in container.list_blobs(name_starts_with=prefix):
+        blob_name = blob_record['name']
+
+        if blob_name.endswith(suffix):
+            blob_list.append(blob_name)
+    return blob_list
+
+
+def download_yamls(account_url: str, container_name: str, credential: str, yaml_urls: List[str],
```

Comment:
Ideally should be iterator, but still with concurrency.

Reply:
This one I wasn't sure how to approach. Are you aware of a example I could model this off?

Reply:
There's a lazy example here: https://github.com/opendatacube/odc-tools/blob/develop/libs/aws/odc/aws/queue.py#L25

The idea is to use the Python 'generator' model, so instead of returning, you `yield` a single item. Just be aware, `yield` is not the same as return, so you need to keep the flow of code happening after.

Homework: https://wiki.python.org/moin/Generators

Reply:
It's easy enough to have `Iterator -> Iterator` for a single-threaded fetch, and it's easy enough to have `List->List` with multi-threading. But having `Iterator->Iterator` with multi-threading is much more involved. So let's just keep it as is, I assume faster fetch for data that fits in memory is more important than dealing with so much data it doesn't fit to RAM.